infer/crates/kanalizer-rs/build.rs

```diff
@@ -76,16 +76,51 @@ fn prepare_huggingface_model() -> anyhow::Result<PathBuf> {
 }
 
 fn download_to(url: &str, path: &Path) -> anyhow::Result<()> {
-    let response = ureq::get(url).call()?;
+    static NUM_ATTEMPTS: usize = 6;
+
     let temp_path = path.with_extra_extension("tmp");
     let mut file = std::fs::File::create(&temp_path)?;
-    std::io::copy(&mut response.into_body().into_reader(), &mut file)?;
-    if path.try_exists()? {
-        std::fs::remove_file(path)?;
+    for i in 0..NUM_ATTEMPTS {
+        let success = download(url, path, &temp_path, &mut file)?;
+        if success {
+            return Ok(());
+        }
+        std::thread::sleep(std::time::Duration::from_secs((i + 1) as u64 * 5));
     }
-    std::fs::rename(&temp_path, path)?;
 
-    Ok(())
+    std::fs::remove_file(&temp_path)?;
+    return Err(anyhow::anyhow!(
+        "Failed to download model after {NUM_ATTEMPTS} attempts"
+    ));
+
+    fn download(
+        url: &str,
+        path: &Path,
+        temp_path: &PathBuf,
+        file: &mut std::fs::File,
+    ) -> anyhow::Result<bool> {
+        let response = ureq::get(url)
+            .config()
+            .http_status_as_error(false)
+            .build()
+            .call()?;
+        if response.status() == 200 {
+            std::io::copy(&mut response.into_body().into_reader(), file)?;
+            if path.try_exists()? {
+                std::fs::remove_file(path)?;
+            }
+            std::fs::rename(temp_path, path)?;
+
+            return Ok(true);
+        }
+        if response.status() == 404 {
+            return Err(anyhow::anyhow!("Model not found: {url}"));
+        }
+        let status = response.status();
+        let body = response.into_body().read_to_string()?;
+        eprintln!("Failed to download model: {status} {body:?}");
+        Ok(false)
```

Comment:
if-else、というよりmatch式の方が見通しが良いような気がしています。

```suggestion
        match response.status().as_u16() {
            200 => {
                std::io::copy(&mut response.into_body().into_reader(), file)?;
                if path.try_exists()? {
                    std::fs::remove_file(path)?;
                }
                std::fs::rename(temp_path, path)?;

                Ok(true)
            }
            404 => Err(anyhow::anyhow!("Model not found: {url}")),
            status => {
                let body = response.into_body().read_to_string()?;
                eprintln!("Failed to download model: {status} {body:?}");
                Ok(false)
            }
        }
```